Lavaland rework

NightDawnFox opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
Своевольный порт лаваленда с йогов со множеством изменений, фич и, конечно же ТУПОРЫЛЫХ БАГОВ.

Снизу представлены все особенности и изменения.

- [x] Добавление на шахту и во все карты нового типа породы - "крепкий базальт". Особенность - требует несколько выстрелов для того, чтобы сломаться. Для Нелли - замаппить необходимые зоны на всех трёх картах.

Представляем вам главную особенность данного ПРа: укрепленная порода!

Отныне вся порода на лаваленде делится на три типа
-Стандартная (старая) порода
-Укрепленная порода
-Лавовый базальт
Важнейшей особенностью и различием этих типов пород является параметр "крепкость". 1 - у стандартной породы, 2 у укрепленной и 3 у базальта. Суть в том, что теперь для уничтожения породы необходимо сделать несколько выстрелов, повредить большим взрывом и так далее. Стандартно, любой предмет снимает одну единицу крепкости, следовательно для уничтожения лавового базальта необходимо сделать три выстрела из стандартной КА/резака

 

https://github.com/ss220-space/Paradise/assets/116907157/0cb0aae4-1e2d-4101-a6d9-013c73454525

Была снижена частота породы, количество ее выпадений. При этом были созданы богатые породой области, при этом уничтожение породы взрывом приносит в 2 раза больше руды, чем простое копание.

![image](https://github.com/ss220-space/Paradise/assets/116907157/a9104256-d3b2-42e3-9ece-fa4c23cd7251)

![image](https://github.com/ss220-space/Paradise/assets/116907157/f8ac7b23-86a9-4700-ba83-f348c52faef1)


- [x] Добавление всех необходимых для нормального функционирования "крепкого базальта" предметов - улучшений для КА, новой взрывчатки для шахты и т.д.

Представляю вам... шахтёрские подрывные заряды!

![image](https://github.com/ss220-space/Paradise/assets/116907157/c2039311-2023-4e5a-84f8-256cd86e0fd2)

Сделанные исключительно для удобной шахтёрской игры, они способны устанавливаться исключительно на породу. На камни Церы тоже нельзя, увы. Заряды не наносят урон и не способны гибнуть, но при этом оглушают, ослепляют и наносят урон ушам в зависимости от мощности заряда и "близости" к взрывному устройству. Так же, добыча с помощью подрывных зарядов приносит дополнительную руду. Берегись!


https://github.com/ss220-space/Paradise/assets/116907157/a4e888a1-c26d-462a-89b2-2764830ba15d

Всего представлено 3 типа взрывных устройств:
-Слабые, за 250 ОДР. Они взрываются на малом расстоянии, относительно бесполезны и малоокупаемы, если не подрывать ими дорогую руду.
 -Средние, за 600 ОДР. Взрываются как на видео, способны достаточно быстро окупить себя и расчистить местность от тяжелых камней.
 -Экспериментальные. По достижению средних уровней технологий (примерно к ЕКА) РНД будет способно печатать шахтёрам крайне мощную и опасную взрывчатку. Она оглушает на длительное время, уничтожает руду в огромном радиусе и полностью окупает себя.
 Был добавлен новый ваучер, в котором содержится 5 малых и 2 средних заряда в сумке для быстрого старта в качестве подрывника. Хороший способ в начале смены достать руду и первые очки.
 В ваучер "Explorer's webbing" была добавлено 2 малых подрывных заряда.

Модуль "укрепления" для КА
Продается в вендоре, занимает 10% емкости КА, цена в 1200 ОДР. Увеличивает силу выстрела КА на 1, позволяя с одного выстрела ломать сильную породу, но для "очень сильной" все еще потребуется два выстрела, если не купить еще одно улучшение..

Большая сумка для руды
Если вы купите ваучер "Plasma Cutter Kit", то помимо резака, вам будет выдана сумка для руды повышенной ёмкости. Разница в 6 дополнительных слотах под руду. Сумка так же была добавлена в вендор как промежуточный этап между  Мсохом за 500 поинтов.

Новая логика выпадения руды.
Отныне вся руда на лаве падает не одним куском, а "кучкой". Это просто визуальное изменение для большего удобства игрокам.
(Картинка кликабельна)
[![noice](https://imgur.com/VpdMdNr.png)](https://www.youtube.com/watch?v=uUdz0AOri4M)

- [x] Порт мировой кузни и связанных с ней предметов и улучшений

Представляем вам... мировую кузню! (старая карта, новая ниже)
![image](https://github.com/ss220-space/Paradise/assets/116907157/34dfa2b9-189a-4f5e-a048-1cbe2d51c375)
Отныне на лаваленде гарантированно спавнится лавовая кузня, с которой могут взаимодействовать шахтёры. К ней всегда ведет сигнал GPS (Tempered signal). Для работы кузни необходим обезвеженный гибтонит и крайне редкий кристалл Магмита. 
Кристалл Магмита можно добыть исключительно из усиленной породы, в обычной он не появляется. При этом, обычное уничтожение породы не оставит от кристалла ни кусочка, для получения кристалла Магмита его нужно добыть с помощью взрыва (гибтонит или шахтёрские подрывные заряды) 

![image](https://github.com/ss220-space/Paradise/assets/116907157/79596a1f-e033-4e03-bf18-dab0fe3ea2af) ![image](https://github.com/ss220-space/Paradise/assets/116907157/d60ddcba-c483-4298-8df8-2b673d4889f0) ![image](https://github.com/ss220-space/Paradise/assets/116907157/46c17758-5220-4847-9949-4cbd853221c9)

После того, как вы получите Магмит, достанете обезвреженный гибтонит и сможете доставить их к кузне, вы сможете расплавить гибтонит для получения "зарядов" кузни. После чего, сможете преобразовать ваш кристалл Магмита в "магмитовые части"
(картинка кликабельна)
[![Я ненавижу гитхаб](https://imgur.com/IAx6oVS.jpg)](http://www.youtube.com/watch?v=Fc-_WNDkodE)

Магмитовые части можно использовать на ЕКА, резаке, веерном резаке и крашере. 
При этом у магмитовых частей есть срок жизни: спустя 10 минут они потухнут и станут непригодными. Тогда их снова придется отнести в кузню и "растопить".
![image](https://github.com/ss220-space/Paradise/assets/116907157/63974be1-853c-485b-af28-11ee08c3cea9)
Подробнее про каждое использование ниже

Магмитовая КА
Отныне ЕКА, которая производится в РНД, имеет 150% вместимости по модкитам.
Если шахтёр хочет вернуть старые значения в виде 200%, то ему нужно немного потрудиться.

Магмитовые резаки. Обладают увеличенной дальностью стрельбы (7 тайлов у улучшенного резака и 6 тайлов у веерного) а так же уникальной особенностью - второе попадание по гибтониту, которое раньше провоцировало взрыв, отныне обезвреживает гибтонит. Помимо этого цвет проджектайла был сменен с фиолетового на красный.

https://github.com/ss220-space/Paradise/assets/116907157/9153a061-f6f1-41b1-8418-ba7159f78a03

Магмитовый Крашер. Помимо прикольного спрайта и измененного цвета проджектайла, он способен пробуриваться всквозь породу наравне с резаками.

![image](https://github.com/ss220-space/Paradise/assets/116907157/85c064e3-2a6d-476e-b43f-74d4d429a573)


- [x] Изменение резаков: переезд обычного резака полностью в вендор и ваучеры, увеличение технологий улучшенного резака, добавление резака-шотгана. Изменение "вместимости" резаков

Отныне стандартный шахтёрский резак можно получить из ваучера в начале смены. Ёмкость обычного резака была сокращена, теперь он требует перезарядки после каждых 30 выстрелов (раньше было 60). РНД более не может делать обычные резаки, они были удалены из протолата. 
![image](https://github.com/ss220-space/Paradise/assets/116907157/4c399567-1f75-43c0-8aa8-dd244795541e)

У продвинутого шахтёрского резака были повышены требуемые технологии. Новые - 5 материалы, 6 плазма, 6 инженерия, 3 комбат и 3 магниты. Были добавлены спрайты "в руках". Понижена ёмкость резака. Теперь он требует перезарядки после 60 выстрелов (раньше было 100)

Новый тип резака: "Индустриальный веерный резак"

![image](https://github.com/ss220-space/Paradise/assets/116907157/b045ed1c-44f8-432f-b73b-7f3b00ae7f9f)

Крайне высокие технологии в РНД: 7 материалы, 5 энергия, 7 плазма, 7 инженерия, 6 комбат и 6 магниты.
Требования в РНД: 6000 металла, 2000 стекла, 2000 плазмы, 2000 золота, 3000 алмазов
Главной особенностью данного резака является веерная стрельба. При стрельбе резак выпускает 5 "пуль", бурящих породу дальностью в 5 тайлов. Разброс - 35. Данные пули не могут отрубить конечность. Ёмкость резака - 20 выстрелов до перезарядки. Главная опасность шахтёров - повышенный шанс на произвольную детонацию гибтонита, ибо безопасное расстояние начинается от 7 тайлов. 
https://www.youtube.com/watch?v=Cd-o6OWdDNg

- [x] Для Нелли: переработать руину с кузней, сделать ее "красиво"

Новая карта мировой кузни:
![image](https://github.com/ss220-space/Paradise/assets/116907157/c93b4733-3475-4536-b4e7-0e17f21d3e75)

- [x] Изменение ТГУИ ОРМ, добавление иконок для сплавов руды

Вот как выглядит текущий ТГУИ ОРМ. 
![image](https://github.com/ss220-space/Paradise/assets/116907157/898bbc17-5fba-4d8e-b8c9-148e40fe2456)
У него есть несколько изъянов: 
-Все сплавы названы максимально странно и неинтуитивно.
-До сплавов необходимо скроллить
-У сплавов некрасивое пустое место после графы "Smelt"
-Да и вообще он какой-то маленький и сплющенный.

А вот как выглядит новый ТГУИ ОРМ!

![image](https://github.com/ss220-space/Paradise/assets/116907157/a770f96d-a6d7-47a9-941e-171490a139f4)
-Большой размер!
-У сплавов теперь большие и красивые иконки!
-Все иконки влезают с учётом всех существующих руд у нас в игре
-У руды теперь указано, какие ресурсы тратятся на создание сплава
ТГУИ был взят с оффов и слегка улучшен под нас (мне не нравятся крошечные размеры их графы sheets, был так же подогнан размер, чтобы было красиво)

- [x] Добавление новых предметов в шахтёрский вендор

В ведор была добавлена новая капсула - "элитный бар". Цена в 10 000 ОДР, добавляет большую капсулу, идеальную для посиделок с товарищами-шахёрами или для расположения в ней передовой культистской базы.
![image](https://github.com/ss220-space/Paradise/assets/116907157/1ef28478-6e15-4cc0-9675-05ed020e0a2e)

Абсолютно новый вид имплантов: импланты, устанавливаемые в ноги!
Для того, чтобы получить какой-либо эффект от имплантов, необходимо установить два одинаковых импланта в каждую ногу. ЭМИ с шансом в 25% наносит серьезный урон ногам (40 брута) и гарантированные 8 брута всегда. Так же оно способно выключать импланты, которые работают не от экшн кнопок.

Добавлен пока единственный ножной имплант: имплант прыгающих ботинок. Отныне шахтёр, желающий получить второй прыжок/использовать связку "прыжковые ботинки+ноуслипы", может заплатить 6000 ОДР, обратиться в лазарет для установки имплантов и пользоваться им.

https://www.youtube.com/watch?v=LIYWGrzAHIU

Были добавлены медицинские мезонные очки - смесь медхудов и мезонных сканеров. Стоимость 1000 ОДР
Добавлены новые карты трансфера очков - за 1000 и 5000 ОДР. 
Добавлен карманный огнетушитель за 200 ОДР
Добавлена возможность покупать сразу 30 маячков освещения. 300 ОДР


- [x] Новые предметы из мегафауны и со шпилей

В лут Пепельному дрейку был добавлен новый предмет "bottle of refined dragons blood".
Его выпивание превращает выпившего в новую расу - драконида.

https://github.com/ss220-space/Paradise/assets/116907157/df8cca50-c3ca-40fc-a034-b9d627f0c7a2

Дракониды - подвид унати, обладающие той же уязвимостью к холоду и восстановлением, что и их собратья, но при этом имеют ряд особенностей:
-Брут мод в 0.8
-Берн мод в 0.9
-Повышенный урон когтями - от 9 до 18
-Повышенный резист к теплу
-Неуязвимость к пепельным бурям
-Повышенная скорость (1.25 от человеческой)
-Невозможность носить ботинки в каком бы то ни было виде (видео не совсем актуально)

Помимо прочего, пеплоходцам и драконидам были вписаны расовые спрайты, теперь пеплоходцы не носят кривые человеские спрайты на одежду и риги

В лут колоссу были добавлены глаза бога. Глазам бога добавлен медицинский худ. Теперь лут из Колосса рандомится - либо глаза, либо голос.

Из шпилей были удалены следующие предметы:
-Глаза бога (перемещены в лут колоссу)
-Диски с модкитами (из-за бесполезности)
-Нульрод священника в виде армблейда (подробнее ниже)

Шпилям были добавлены следующие предметы:
-Рандомный нульрод священника - шансы остаются те же, что и у армблейда, просто теперь вместо гарантированного определенного нульрода рандомится любой
-Лечащие гланды обдукторов (их необходимо установить в лазарете или сбросить в РНД)
-Кусок мифрила (используется в крафте, про него - в минорных изменениях)
-Букет усиленных цветов - подробнее ниже
-Рунный скимитар - [хороший](https://oldschool.runescape.wiki/w/Rune_scimitar#Item_sources) меч с уроном в 28 единиц
-Набор из инъектора дварфизма и трёх экспериментальных подрывных зарядов
-Случаный магмитовый шахтёрский инструмент

Букет усиленных цветов - букет, который способен при использовании и нахождении полторы секунды у фауны, "утихомирить" любую фауну на лаваленде, сделав ее другом. Фауна поменяет цвет и будет миролюбивой. Игрок способен призвать всю* прирученную фауну при использовании букета в руках, однако: 
-На z-уровне станции призвать фауну нельзя
-Мегафауна не призывается
-На любом другом уровне фауну можно призвать только "на улице", вне помещений

https://www.youtube.com/watch?v=Bus-6PCnujA

Мегафауна тоже может стать миролюбивой, однако для нее необходимо 35 раз использовать букет цветов вплотную, что технически не совсем возможно, если мы говорим про колосса или баблгама, но всё равно забавно. Прирученная магафауна выронит сундук с лутом, как она это делает в случае победы над ней

![image](https://github.com/ss220-space/Paradise/assets/116907157/0913366a-3e01-4543-86fa-d59b568c4a66)

- [x] Иные минорные изменения - новая фауна, возможность разделывать златожоров, иное.

Ну, поехали.

Была добавлена новая тварь на лаваленд - марроу уивер. 

![image](https://github.com/ss220-space/Paradise/assets/116907157/e887e18b-5804-4874-9105-7f248a34acd8)

Из особенностей:

При укусе впрыскивает яд - споры, что замыливают экран.
Если нанести вред и умереть от нее, то она сожрет все внутренние органы в торсе
При снятии более 2/3 здоровья становится агрессивнее, наносит больше урона, бегает быстрее, начинает впрыскивать "venom" - один из самых опасных ядов в игре

Примеры тут:
https://discord.com/channels/617003227182792704/928706798670721145/1131551576092332122

Теперь златожоров можно разделать. При убийстве из них выпадает "плоть златожора" (спрайт за авторством Петра). Лечит брут и берн в огромных количествах, но низкий порог передозировки и ядовитая сырой. Можно пожарить в лаве, тогда яд исчезнет. Златожоры теперь активнее едят руду и способны сохранить в себе до 15 единиц руды

![image](https://github.com/ss220-space/Paradise/assets/116907157/38f347c8-1cfa-47a5-80c8-460e7637e4f7)


Добавлена новая одежда:
-Обертки для ног из голиафа
-Обертки для ног из дрейка
-Костяной лук и стрелы из костей (из костяного лука можно стрелять только костяными стрелами, из обычного только обычными)
-Перчатки из шкуры легиона
-Перчатки из хитина вивера
-Костяной меч
-Костяной щит
-Хитиновое копье
-Хитиновый колчан (в отличие от обычного, находится на поясе)
Теперь все колчаны показывают стрелы цифрами.
-Броня первопроходца. Резист к буре, возможность улучшать плитами голиафов

![image](https://github.com/ss220-space/Paradise/assets/116907157/4c328920-0f20-4de9-9712-499dce00c72f)



Твики пеплоходцев:
Добавлен новый подтип пеплоходцев - шаманы пеплоходцев.

![image](https://github.com/ss220-space/Paradise/assets/116907157/5923c85b-8a32-4c9c-877d-40aa679a1d45)

"В племени" может одновременно находиться исключительно один шаман. Пока тело старого находится в игровой области (мертв, но не гибнут), новый шаман не появится. Гораздо слабее, чем обычные пеплоходцы, но при этом имеет возможность лечить других людей магической способностью (себя лечить не может). Появляется сразу в броне и в новой одежде, которую не может надеть никто, кроме унати и его подтипов.

![image](https://github.com/ss220-space/Paradise/assets/116907157/2c0030e5-b557-4e7d-8eb7-67ea965edc34)

Теперь трофей с иерофанта создает гончих вместо стены. Немного ослаблен колосс, чтобы его можно было убивать с Крашера


https://github.com/ss220-space/Paradise/assets/116907157/d639aead-fe15-47be-99c9-7a74c13e23d9


Теперь у шахтеров раундстартом не автоматический сканер, а ручной. Плохой автоматический сканер продается в вендоре за 800 поинтов, улучшенный за 3000 поинтов. Сканеры теперь издают звуки! Отключаемо через альтклик.

https://youtu.be/prLqVL-u4Xc

Новые спрайты для костяного копья, для костяного топора, для ручного сканера, для обычного и улучшенного сканера.

![image](https://github.com/ss220-space/Paradise/assets/116907157/22f6da69-1bea-45e4-b36a-fee1a3603abc)
![image](https://github.com/ss220-space/Paradise/assets/116907157/50813783-227c-44c5-8c14-6c255b5b139a) 
![image](https://github.com/ss220-space/Paradise/assets/116907157/2b1f6d1b-ab39-446f-bdc0-c3736ebc109b)




Уменьшены шансы на породу вообще везде, чтобы жизнь мёдом не казалась
Уменьшено количество выпадающей породы отовсюду.
При этом, "взрыв" породы, вызванный гибтонитом и подрывной взрывчаткой, увеличивает количество выпадаемой руды в 2 раза.
Общие цифры:
обычная порода - от 1 до 2
Укрепленная - от 1 до 4
Усиленная - от 1 до 6
Всё это с возможностью удвоить цифры, если использовать взрывчатку (подрывные заряды или гибтонит)

Увеличен бюджет лавы на 10 единиц.
Добавлено 5 новых лавовых руин
Изменено 6 руин за авторством Нелли и nohumanityok (ливнул с проекта, восстанавливал по картинкам)

Новые руины:

Биокупол "Озеленение"

Это - один из биокуполов для жилья, разработанный корпорацией NanoTrasen совместно с дионами. Данный биокупол не предназначался для Лаваленда. Как он здесь оказался - неизвестно. Возможно, очередная блюспейс-аномалия, или вовсе что-то другое.

![image](https://github.com/ss220-space/Paradise/assets/116907157/c95fd210-447a-47c6-86e5-0ce1099a9282)

Логово дракона

Гнездо вершины пищевой цепочки лаваленда - пепельного дракона. Будучи самым опасным представителем своего вида, окруженный останками своей добычи, он охраняет древние сокровища лаваленда

![image](https://github.com/ss220-space/Paradise/assets/116907157/5e292607-25a7-47eb-b3ce-bc2e5a4e3708)


Последнее прибежище

Место упокоения великого война и охотника. Отныне лишь древние руины и бездвижные могилы напоминают о нём.
Особенность - тут расположена уникальная катана, которая имеет те же характеристики, что и обычная, но наносит увеличенный урон фауне лаваленда. При попытке вытащить катану, из рядом находящихся могил вылезут погребенные (легионы и скелеты)

![image](https://github.com/ss220-space/Paradise/assets/116907157/14137464-d9a5-4e85-a985-a6bc9bc0ef4e)


Гнездо мароу уиверов

Место скопления одной из самой опасой фауны лаваленда - ядовитых пауков, впрыскивающих яд при каждом укусе. Опасные сами по себе, они будут драться как загнанные в угол, если кто-то попробует покуситься на их гнездо и добычу внутри. И вероятнее всего эта встреча закончится не в пользу вторженца.

![image](https://github.com/ss220-space/Paradise/assets/116907157/6e86d2e1-5bc6-4afe-af2b-cbc7b8ebf52a)

Баракки "Легионеров"

Последнее пристанище Легиона. Теперь эти души вечно скитаются по пустошам в поисках новых жертв. 

![image](https://github.com/ss220-space/Paradise/assets/116907157/c0c32d0d-0679-47ee-8f59-aa9329cb03dd)


Переделанные руины:

Был слегка улучшен бар:

![image](https://github.com/ss220-space/Paradise/assets/116907157/f36b05f2-a9d5-481c-b521-f8c7c0d1d918)

Переделан зимний биодом:

![image](https://github.com/ss220-space/Paradise/assets/116907157/848ed7d5-f9ba-4716-af83-eb1e5522d376)

Зависть обрела новый вид:

![image](https://github.com/ss220-space/Paradise/assets/116907157/068812bf-993a-4752-9324-fbaafaf6bbd6)

Холл с лечащим фонтаном теперь выглядит свежо и по-новому!

![image](https://github.com/ss220-space/Paradise/assets/116907157/b56a6dba-b866-4716-8ecc-86794e76168c)

Големы сдали в металлолом свой древний корабль и приобрели новый шахтерский фрегат, разработанный НТ:

![image](https://github.com/ss220-space/Paradise/assets/116907157/92423d23-b0a9-463d-a0d6-769c1b67616f)


мне как-то лень расписывать про эту руину:

![image](https://github.com/ss220-space/Paradise/assets/116907157/1bb1455d-c2f6-4ab1-a3cb-4abda13ecc90)

Часть этих карт была портирована с Йогов, часть была восстановлена и взята за авторством "nohumanityok", что, к сожалению, покинул наш проект и Нелли, что занимался некоторыми картами, спасибо ему

- [x] Перенести гемы из отдельного ПРа в общий ПР по реворку фауны

Мне в лом дублировать документ, гляньте тут всё:
https://discord.com/channels/617003227182792704/708456634468270100/1125762903098802277

Мне даже жутко представить, сколько тут багов и проблемного кода..
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
https://discord.com/channels/617003227182792704/755125334097133628/1135559498124234892
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

NightDawnFox committed:
Пока так

Брошено на полпути, Нелли обещал перехватить для того, чтобы замаппить участки с твёрдой породой

captainnelly commented:
это база

NightDawnFox committed:
Обновление

Новая руина+карта
Куча спрайтов для будущих предметов
Механика крафта запчастей из магмита

NightDawnFox committed:
Угадайте, кто говнокодер?

NightDawnFox committed:
Обновление 2

Добавлен "Plasma cutter shotgun", способный быстро уничтожать породу за безумно высокие технологии
Обычный резак полностью переведён из РНД в вендор, повышены технологии для улучшенного резака, добавлен новый ваучер с резаком, 30 маркеров и ручной огнетушитель в вендор

NightDawnFox committed:
Обновление 3

Наковальня теперь проигрывает звуки. ЕКА теперь обладает 150% вместимостью вместо 200%.
Добавлена Mega KA со вместимостью в 200% Для получения нужно использовать части Магмита на ЕКА.
Новый вар, отвечающий за силу, с которой стреляет КА по породе Новый модкит, увеличивающий силу стрельбы по породе, добавление модкита с вендор

NightDawnFox committed:
С запятыми с детства я дружу

NightDawnFox committed:
500 исправлений сделаны, 1000 уже на подходе

NightDawnFox committed:
Update lavaland_surface_worldanvil.dmm

NightDawnFox committed:
Обновление 4

Добавлено 3 новых применения для частей магмита:
-Магмитовый резак
-Магмитовый веерный резак
-Магмитовый крашер.
Резаки стреляют дальше и автоматически обезвреживают гибтонит при стрельбе
Крашер стреляет марками так же, как и резаки

NightDawnFox committed:
балансировка чисел

NightDawnFox committed:
Обновление 5

Добавлена улучшенная сумка для руды. Емкость - 16 ячеек (у стандартной 10). Покупается в вендоре за 500 поинтов.
Добавлены взрывные заряды!
Особенность взрывных зарядов в том, что они способны взрывать исключительно породу. Взрыв породы с рудой заставляет оную выронить в 2 раза больше ресурсов.
Есть 3 типа зарядов - слабые, стандартные и усиленные. Усиленные - только в РНД

NightDawnFox committed:
Ни дня без мгновенных балансных правок.

NightDawnFox committed:
Обновление 5.5

Небольшой багфикс и...
Изменена логика стаков у руды. Отныне руда падает не единым куском, а "кучкой", что приятно влияет на погружение

NightDawnFox committed:
Обновление 6

Изменение ТГУИ у ОРМ:
Он стал больше!
Теперь у справов "нормальное", игровое название
Добавлено описание, что нужно для сплавов
Изменение столбов в более приятную форму

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Исправление мёрдж конфликтов

NightDawnFox committed:
Обновление 7

Добавлены 2 новых трансферных карты - на 1000 и на 5000 очков.
Добавлена новая капсула - элитная барная капсула с баром. Цена - 15000
Добавлен новый сканер - медицинские мезонные очки.
Новый тип имплантов: импланты, устанавливаемые в ноги. Они начинают работать только в случае, если будут установлены в обе.. ноги. Воздействие ЭМИ с шансом в 25% вызывает 40 брута в конечность. Так же предусмотрен вывод импланта из строя на некоторое время. Добавлен имплант прыгающих ботинок в шахтёрский вендор.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Багфикс

Исправление капсулы, добавление ей "tiny fan" у входов в капсулу. Уменьшение цены до 10000

NightDawnFox committed:
Обновление 8

Новая подраса - Дракониды!
Как унати, только лучше. Падают из Дрейка. Слегка быстрее, держат больше урона, имунны к буре и устойчивы к теплу.
Исправлен визуал у пеплоходцев, теперь расовые спрайты и диги-одежда отображается на них нормально

NightDawnFox committed:
Обновление 9

Теперь глаз бога - лут Колосса
У глаза бога теперь встроенный медхуд.
Был добавлен мифрил как минерал. Пока он "просто существует"
Изменение лута с тендриллов:
Удалены диски с модкитами
Удален глаз бога
Теперь нулльрод вместо гарантированного "армблейда", рандомится на любой нульрод священника
Добавлены абдукторские лечащие гланды, которые еще нужно поставить в лазарете
Добавлен мифрил
Добавлены "цветы" в лут.
Если ударить им фауну, она становится мирной и начинает сражаться за тебя. Фауну можно призвать (только на лаве и на открытом воздухе)
Мегафауну тоже можно приручить, но оно требует 35 ударов по фауне, что не совсем реально

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Обновление 10

Немного разбавлен лут из шпилей,
Добавлен рунный скимитар с уроном 28,
Добавлен набор с инъектором дварфмизма и тремя мощными подрывными зарядами
Добавлена случайная мега-вариация шахтерских инструментов

NightDawnFox committed:
Обновление 11

Теперь златожоров можно разделать.
Теперь златожоры могут есть титан.
У златожоров улучшен поиск руды для питания
Теперь златожоры могут есть до 15 единиц руды, улучшен код их "поедания"
Иные изменения златожоров

NightDawnFox committed:
Обновление 12

Добавлен абсолютно новый моб - марроу вивер. При укусе впрыскивает яд. Если ранен, то пожирает внутренние органы в торсе у убитой жертвы. Если здоровья осталось менее 1/3, то становится агрессивнее и впрыскивает более опасный яд. При разделке дает новый предмет - хитин, требующийся для крафта новых предметов.
Куча новых предметов для крафта:
-Обертки для ног из голиафа и дрейка. Дают броню
-Перчатки из голиафа и хитина вивера. Перчатки из хитина дают повышенный шанс уронить. Перчатки из голиафа дают броню
-Меч из костей. 18 урона, 15 бронебития, 30 шанса блокировать урон
Щит из голиафа - 30 шанса блокировать урон
Копье из хитина - чутка улучшенное по характеристикам обычное копье
Костяной лук - использующий специальные костяные стрелы лук, не имеет замедления, в отличие от обычного лука.
Костяные стрелы - улучшенные стрелы, которые можно использовать только в костяном луке. Наносят 45 брута, -10 бронепробития
Хитиновый колчан - колчан, который ставится в слот пояса, больше места, чем в обычном колчане
Заделка под новую броню.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

TrashDoxx commented:
Какая ж тупая бессмысленная хуйня, которая не решает абсолютно ничего.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
я не знаю, что я делаю

NightDawnFox committed:
Обновление 13

Шаманы пеплоходцев!
Теперь первое яйцо в гнезде пеплоходцев спавнит уникального - шамана пеплоходцев, умеющего лечить раненных с помощью способности. Одновременно на карте может быть только один шаман.

NightDawnFox committed:
Обновление 13.5

Изменение принципа работы вортекс талисмана: теперь вместо создания стенки, он создаёт гончих после удара по метке.

NightDawnFox committed:
Обновление 14

Новая крафтовая броня - броня первопроходца. Особенности:
 - ужасающе дорогой крафт, требующий 7 шкур голиафа, 8 костей, 10 жил и 10 хитина.
 - Высокие показатели брони после всех улучшений (броня улучшается шкурами голиафа)
 - полная защита от бурь
Ну и всё, в целом..

NightDawnFox committed:
Финальное обновление перед заливом ДММ файлов

Теперь вместо кишок златожоров вываливается мясо, которое можно пожарить по аналогии с мясом голиафа. Сырое мясо токсично, но при этом лечит брутоберн. Пожаренное - хороший источник лечения

NightDawnFox committed:
Балансная правка

NightDawnFox committed:
Всякий хлам для маппинга

Да.

NightDawnFox committed:
Маппинг!

NightDawnFox committed:
Твики лавовой породы

Налажена работоспособность потрескавшейся породы.
Уменьшены шансы на породу вообще везде, чтобы жизнь мёдом не казалась
Уменьшено количество выпадающей породы отовсюду.
При этом, "взрыв" породы, вызванный гибтонитом и подрывной взрывчаткой, увеличивает количество выпадаемой руды в 2 раза.
Общие цифры:
обычная порода - от 1 до 2
Укрепленная - от 1 до 4
Усиленная - от 1 до 6 с возможностью удвоить цифры, если использовать взрывчатку.
Починена чёрная руда.
Приведены названия в порядок. Никаких более "rock" для всей породы с минералами

NightDawnFox committed:
Хлам для маппинга

Я устал расписывать...

NightDawnFox committed:
обновление 242354

Еще немного хлама для маппинга
Небольшой нерф для Колосса, дающий возможность убивать его крашером после удаления стенки
Твики сканеров!
Отныне у шахтеров раундстартом спавнится не автоматический сканер, а ручной! Слабый автоматический сканер теперь продается в вендоре по стоимости как улучшенный. А улучшенный продаётся за 3к поинтов
У сканеров теперь есть звуки работы! Отключаемые.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Маппинг 2!

Увеличен бюджет лавы на 10 единиц.
Добавлено 5 новых лавовых руин
Изменено 6 руин за авторством Нелли и nohumanityok (ливнул с проекта, восстанавливал по картинкам)

NightDawnFox committed:
Фиксы

NightDawnFox committed:
фиксы 2

NightDawnFox committed:
Фиксы 3

NightDawnFox committed:
ДА КАК Я ВООБЩЕ СМОГ ДВА ТЮРФА В ОДИН ТАЙЛ ЗАПИХАТЬ

NightDawnFox commented:
Надо потихоньку заканчивать, пока я не выгорел. Остались лишь гемы...

NightDawnFox committed:
Последнее (я надеюсь) обновление

NightDawnFox committed:
Исправление маппы...

TrashDoxx commented:
А мне похуй. Удаляй.

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
фикс тгуи

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Да

NightDawnFox committed:
Update unathi.dm

Фиксим проблемы с совместимостью

NightDawnFox committed:
Фиксы багов или типа того

1. Починил шаттл големов, чтобы его не разрывало на части в космосе
2. Убрал из лута мифрил (оставлен случайно)

Den109G commented on code/modules/mining/equipment/upgrades.dm:
```diff
@@ -0,0 +1,84 @@
+	addtimer(CALLBACK(src, PROC_REF(go_inert)), 10 MINUTES)
+
+/obj/item/magmite_parts/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(inert)
```
Нету проверки на `proximity_flag`. Очень интересно наверное красть КА у подноса учёного.

Den109G commented on _maps/map_files/RandomRuins/SpaceRuins/ussp.dmm:
```diff
@@ -2484,7 +2484,7 @@
 	name = "suspicious button"
 	},
 /obj/machinery/vending/cigarette/free{
-	slogan_list = list("Just remember! No capitalist.","Best enjoyed with Vodka!.","Smoke!","Nine out of ten USSP scientists agree, smoking reduces stress!","There's no cigarette like a Russian cigarette!","Cigarettes! Now with 100% less capitalism.")
```
Тут пробелы поехали.
![image](https://github.com/ss220-space/Paradise/assets/87372121/0f8a48e2-ac85-4672-a5e6-9a2a337a29d3)

Den109G commented:
Консоль големов призыва шаттла не работает, потому что стоит на улице не в зоне.
А ловко ты это придумал.

Den109G commented:
Есть причины на руине ставить нерушимые стены? Чтобы не сломали камень и не увидели там воды?
Особенно придавая им вид камня?
![image](https://github.com/ss220-space/Paradise/assets/87372121/07796f78-0eec-4588-9a8c-b0f6d674535e)

Den109G commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -231,3 +243,76 @@
+		if(M.sentience_type != SENTIENCE_ORGANIC)
+			to_chat(user, span_warning("[M] cannot be tamed!"))
+			return
+		if(!do_after(user, 1.5 SECONDS, M))
```
Где-то тут не показывает полоску.
Скорей всего ошибка.
```suggestion
		if(!do_after(user, 1.5 SECONDS, target = M))
```

Den109G commented on code/modules/mining/equipment/kinetic_crusher.dm:
```diff
@@ -476,3 +475,52 @@
+		var/turf/proj_turf = user.loc
+		if(!isturf(proj_turf))
+			return
+		var/obj/item/projectile/destabilizer/mega/D = new /obj/item/projectile/destabilizer/mega(proj_turf)
```
Это же просто поменять снаряд, да?
Весь этот код `afterattack` я бы поменял на тип дестабилизатора у кинетика и просто поменял бы их.
тип
```
/obj/item/twohanded/kinetic_crusher
   var/obj/item/projectile/destabilizer/destab = /obj/item/projectile/destabilizer
   
/obj/item/twohanded/kinetic_crusher
   destab = /obj/item/projectile/destabilizer/mega
   
/obj/item/twohanded/kinetic_crusher/mega/afterattack
   new destab()
 ```

Den109G commented on code/modules/surgery/surgery.dm:
```diff
@@ -102,6 +102,9 @@
 	if(tool)
 		speed_mod = tool.toolspeed * gettoolspeedmod(user)
 
+	if(is_species(user, /datum/species/unathi/ashwalker/shaman))//shaman is slightly better at surgeries
```
Скорость от расы и так от if-а выше смотрится.
Или специально для хирургии, у шамана скорость не 0.9 , а 0.9*0.9 (0.81)?

NightDawnFox replied:
Да, такова идея (так было на йогах, и, я думаю, это их фича)

Den109G reviewed (commented):
Все файлы мельком просмотрел. В основном, 90-100 файлов - это dmi, dmm и изменения sprite_sheets
Основные замечания сказал. Остальное - дело баланса.

NightDawnFox committed:
Исправления в связи с ревью

1. Поправлены поплывшие фразы у СССП
2. Исправлено зонирование на шаттле големов
3. Переписан код крашера для более удобного использования
4. Добавлена проверка на "proximity flag" для деталей магмита
5. Исправлена проверка на "target", чтобы появлялась полоска

NightDawnFox commented:
@Den109G Честно говоря, нерушимые стены нужны для того, чтобы сохранялся визуальный стиль и форма биодома, в котором находятся гигантские горы. Ну и было бы странно ломать то, откуда "течёт" река. Всё остальное поправил и улучшил с учётом замечаний.
А еще я просто взял стены с карты с ниндзей, они там такие...

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

SAADf603 commented on _maps/map_files/RandomRuins/LavaRuins/lavaland_surface_dragon.dmm:
три халявных сундука - многовато для зоны, которую можно обойти без конфронтации с медленными мобами.
один ещё куда не шло, но не три.

NightDawnFox replied:
Поправил

NightDawnFox committed:
Исправления по приказу глав.маппера

"Один - еще куда не шло.."

NightDawnFox committed:
Еще один ряд правок

1. Исправлен баг с биодомом зимы
2. Улучшен фонтан, обычная порода частично заменена на укрепленную
3. Улучшена зависить, порода была частично заменена на укрепленную

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Пересборка тгуи. Ммммм

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
фикс конфликтов

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into better-lava

NightDawnFox committed:
Я люблю пересобирать тгуи

Dimach commented on code/modules/projectiles/guns/projectile/bow.dm:
```diff
@@ -84,24 +107,49 @@
 	for(var/i in 1 to storage_slots)
 		new /obj/item/ammo_casing/caseless/arrow(src)
-	update_icon()
+		update_icon()
```
```suggestion
	update_icon()
```

Dimach committed:
Update code/modules/projectiles/guns/projectile/bow.dm

Dimach committed:
Merge branch 'master220' into better-lava

# Conflicts:
#	tgui/packages/tgui/public/tgui.bundle.js

Dimach committed:
Rebuild tgui

TrashDoxx commented:
Что ж, пора теперь портировать это: https://github.com/tgstation/tgstation/pull/78040

TrashDoxx commented:
))

Koko4yXa commented:
не уверен тут это менялось или нет, но карта на трансфера 1000 поинтов в автомате выдает карту на 100 поинтов, верните нолик
